scope/src/doctor/check.rs

```diff
@@ -289,17 +304,38 @@ impl DefaultDoctorActionRun {
         }
     }
 
-    async fn run_fixes(&self) -> Result<(i32, Vec<ActionTaskReport>), RuntimeError> {
-        let mut action_reports = Vec::new();
-        let mut highest_exit_code = -1;
-        if let Some(action_command) = &self.action.fix.command {
-            for command in &action_command.commands {
-                let report = self.run_single_fix(command).await?;
-                highest_exit_code = max(highest_exit_code, report.exit_code.unwrap_or(-1));
-                action_reports.push(report);
-                if highest_exit_code >= 100 {
-                    return Ok((highest_exit_code, action_reports));
+    async fn run_fixes(
+        &self,
+        prompt: for<'a> fn(&'a str) -> bool,
+    ) -> Result<(i32, Vec<ActionTaskReport>), RuntimeError> {
+        match &self.action.fix.command {
+            None => Ok((NO_COMMANDS_EXIT_CODE, Vec::new())),
+            Some(action_command) => match &self.action.fix.prompt {
+                None => Ok(self.run_commands(&action_command.commands).await?),
+                Some(prompt_text) => {
+                    if prompt(&prompt_text) {
+                        Ok(self.run_commands(&action_command.commands).await?)
+                    } else {
+                        Ok((USER_DENIED_EXIT_CODE, Vec::new()))
+                    }
                 }
+            },
+        }
+    }
+
+    async fn run_commands(
+        &self,
+        commands: &Vec<String>,
+    ) -> Result<(i32, Vec<ActionTaskReport>), RuntimeError> {
+        let mut action_reports = Vec::new();
+        let mut highest_exit_code = NO_COMMANDS_EXIT_CODE;
+
+        for command in commands {
+            let report = self.run_single_fix(command).await?;
+            highest_exit_code = max(highest_exit_code, report.exit_code.unwrap_or(NO_COMMANDS_EXIT_CODE));
```

Comment:
I think maybe the `-1` exit code was being overloaded.  
Should we add another exit code?